src/intervals/arithmetic/basic.jl

```diff
@@ -0,0 +1,302 @@
+
+Implement the `div` function of the IEEE Std 1788-2015 (Table 9.1).
+
+Note: the behavior of the division is flavor dependent for some edge cases.
+"""
+function /(a::F, x::Real) where {F<:Interval}
+    isempty(a) && return emptyinterval(T)
+    iszero(x) && return div_by_thin_zero(a)
+
+    if x ≥ 0.0
+        return @round(F, a.lo/x, a.hi/x)
+    else
+        return @round(F, a.hi/x, a.lo/x)
+    end
+end
+
+/(x::Real, a::Interval) = x*inv(a)
+
+function /(a::F, b::F) where {T, F<:Interval{T}}
+    (isempty(a) || isempty(b)) && return emptyinterval(F)
+    isthinzero(b) && return div_by_thin_zero(a)
+
+    if b.lo > zero(T) # b strictly positive
+        a.lo >= zero(T) && return @round(F, a.lo/b.hi, a.hi/b.lo)
+        a.hi <= zero(T) && return @round(F, a.lo/b.lo, a.hi/b.hi)
+        return @round(F, a.lo/b.lo, a.hi/b.lo)  # zero(T) ∈ a
+
+    elseif b.hi < zero(T) # b strictly negative
+        a.lo >= zero(T) && return @round(F, a.hi/b.hi, a.lo/b.lo)
+        a.hi <= zero(T) && return @round(F, a.hi/b.lo, a.lo/b.hi)
+        return @round(F, a.hi/b.hi, a.lo/b.hi)  # zero(T) ∈ a
+
+    else   # b contains zero, but is not zero(b)
+        isthinzero(a) && return a
+
+        if iszero(b.lo)
+            a.lo >= zero(T) && return @round(F, a.lo/b.hi, T(Inf))
+            a.hi <= zero(T) && return @round(F, T(-Inf), a.hi/b.hi)
+            return entireinterval(F)
+
+        elseif iszero(b.hi)
+            a.lo >= zero(T) && return @round(F, T(-Inf), a.lo/b.lo)
+            a.hi <= zero(T) && return @round(F, a.hi/b.lo, T(Inf))
+            return entireinterval(F)
+
+        else
+            return entireinterval(F)
+
+        end
+    end
+end
+
+"""
+    inv(a::Interval)
+
+Implement the `recip` function of the IEEE Std 1788-2015 (Table 9.1).
+
+Note: the behavior of this function is flavor dependent for some edge cases.
+"""
+function inv(a::F) where {T, F<:Interval{T}}
+    isempty(a) && return emptyinterval(F)
+
+    if zero(T) ∈ a
+        a.lo < zero(T) == a.hi && return @round(F, T(-Inf), inv(a.lo))
+        a.lo == zero(T) < a.hi && return @round(F, inv(a.hi), T(Inf))
+        a.lo < zero(T) < a.hi && return entireinterval(F)
+        isthinzero(a) && return div_by_thin_zero(one(F))
+    end
+
+    return @round(F, inv(a.hi), inv(a.lo))
+end
+
+# Rational division
+//(a::Interval, b::Interval) = a / b
+
+function min_ignore_nans(args...)
+    min(Iterators.filter(x->!isnan(x), args)...)
+end
+
+function max_ignore_nans(args...)
+    max(Iterators.filter(x->!isnan(x), args)...)
+end
+
+"""
+    fma(a::Interval, b::Interval, c::Interval)
+
+Fused multiply-add.
+
+Implement the `fma` function of the IEEE Std 1788-2015 (Table 9.1).
+"""
+function fma(a::F, b::F, c::F) where {T, F<:Interval{T}}
+    (isempty(a) || isempty(b) || isempty(c)) && return emptyinterval(F)
+
+    if isentire(a)
+        isthinzero(b) && return c
+        return entireinterval(F)
+    elseif isentire(b)
+        isthinzero(a) && return c
```

Comment:
This should perhaps use `zero_times_infinity` too.

Reply:
fwiw, I had updated fma not to use setrounding in another PR